### PR TITLE
Referenced methods #17

### PR DESCRIPTION
**Pull request by stachu540**
### Prerequisites
* [x] I have read and understood the [contribution guidelines](CONTRIBUTING.md)
* [x] This pull request follows the code style of the project
* [x] I have tested this feature **_Reviewing Required!_**

### Changes Proposed
* Methods added on `AbstractChannelEvent` #17 (you can now interact methods extended this class)
* Added `final` keyword to preventing data changes around `Event`'s
* Added **private** methods into `AbstractChannelEvent` when checking bot it is moderator (via `ChannelCache`) and channel editor (via API)

### Additional Notes
Now you can use methods inside Event emitters. I think it is all changes. Ready to roll.

**Question**: Can you clarify, do you want make a PubSub Listener? On separate PR of course.

### CODE REVIEW

**Comment by PhilippHeuer** [comment]
I like the added methods in the channel endpoint, it makes responding to events easier for bots.

And yes we should add pubsub listeners since that is the best way to listen for events (more reliable than irc parsing/...)

File: src/main/java/me/philippheuer/twitch4j/events/event/irc/ChannelLeaveEvent.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
 	/**
 	 * User
 	 */
-	private User user;
+	private final User user;
```
[the remark this answers is not in the document]
Comment (stachu540) [reply src/main/java/me/philippheuer/twitch4j/events/event/irc/ChannelLeaveEvent.java]: Reverting commit `b35acfd`

**Comment by stachu540** [comment]
Will do next PR

**Comment by PhilippHeuer** [comment]
Looks good now ;)

**Pull request merged**